### PR TITLE
simplify HTTPUrlConnection matchers

### PR DESCRIPTION
**Pull request by richardstartin**
The matcher here actually specified the contents of some packages, so I just took the class names from those packages.

### CODE REVIEW

File: dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java [hunk 1]
```diff
@@ -35,11 +30,11 @@ public HttpUrlConnectionInstrumentation() {
...
-    return nameStartsWith("java.net.")
-        .or(ElementMatchers.<TypeDescription>nameStartsWith("sun.net"))
-        // This class is a simple delegator. Skip because it does not update its `connected` field.
-        .and(not(named("sun.net.www.protocol.https.HttpsURLConnectionImpl")))
-        .and(extendsClass(named("java.net.HttpURLConnection")));
+    return namedOneOf(
+        "sun.net.www.protocol.https.HttpsURLConnection",
+        "sun.net.www.protocol.http.HttpURLConnection",
+        "java.net.HttpURLConnection",
+        "sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection");
```
Comment (richardstartin) [inline dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java:37]: This previously matched, but we could ignore it safely.

File: dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java [hunk 2]
```diff
@@ -35,11 +30,10 @@ public HttpUrlConnectionInstrumentation() {
...
   public ElementMatcher<TypeDescription> typeMatcher() {
-    return nameStartsWith("java.net.")
-        .or(ElementMatchers.<TypeDescription>nameStartsWith("sun.net"))
-        // This class is a simple delegator. Skip because it does not update its `connected` field.
-        .and(not(named("sun.net.www.protocol.https.HttpsURLConnectionImpl")))
-        .and(extendsClass(named("java.net.HttpURLConnection")));
+    return namedOneOf(
+        "sun.net.www.protocol.https.HttpsURLConnection",
+        "sun.net.www.protocol.http.HttpURLConnection",
+        "java.net.HttpURLConnection");
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java:36]: Does IBM have a different name for these? (eg `javax.net.ssl.HttpURLConnection`)

**Review by tylerbenson — APPROVED** [review]
seems to have passed the IBM test, so I guess this is fine.

File: dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java (hunk 2 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/http-url-connection/src/main/java/datadog/trace/instrumentation/http_url_connection/HttpUrlConnectionInstrumentation.java:36]: If you look at the matcher this replaces, it only matches sun.net and java.net packages

**Review by bantonsson — APPROVED** [review]

**Pull request merged**